src/components/app/App.jsx

```diff
@@ -92,22 +93,40 @@ const queryClient = new QueryClient({
   },
 });
 
-const Root = () => (
-  <NoticesProvider>
-    <ToastsProvider>
-      <Toasts />
-      <Suspense fallback={<DelayedFallbackContainer />}>
-        <Outlet />
-      </Suspense>
-    </ToastsProvider>
-    <ScrollRestoration />
-  </NoticesProvider>
-);
+const Root = () => {
+  const navigation = useNavigation();
+  const fetchers = useFetchers();
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      const fetchersIdle = fetchers.every((f) => f.state === 'idle');
+      if (navigation.state === 'idle' && fetchersIdle) {
+        NProgress.done();
+      } else {
+        NProgress.start();
+      }
+    }, 300);
```

Comment:
[nit] Did we want to abstract that as a constant

Reply:
Sure, can do that before merge! Good suggestion.